pkg/controllers/dashboard/controller.go

```diff
@@ -72,6 +74,19 @@ func Register(ctx context.Context, wrangler *wrangler.Context, embedded bool, re
 		}
 	}
 
+	// In the case where Rancher is embedded and running in the Harvester local cluster,
+	// we need to manage the SystemUpgradeControllerReady condition for the local cluster.
+	if !features.MCMAgent.Enabled() && !features.MCM.Enabled() && features.Harvester.Enabled() {
```

Comment:
Would it make more sense to call `Register` here with a config object populated with these values (instead of exporting them)?

So something like
```
rkecontrolplanecondition.Register(ctx, &config.UserContext{
  ClusterName: "local",
  Management: &config.ManagementContext{
    Wrangler: &wrangler.Context{
    // so forth an so on
    },
  },
})
```
Typically the register functions take the place of create handlers raw, so if not I'd suggest just passing each variable (name, caches, & clients) in the `Register` function.